core/src/main/scalajvm/sttp/client4/internal/httpclient/BodyToHttpClient.scala

```diff
@@ -46,10 +45,17 @@ private[client4] trait BodyToHttpClient[F[_], S, R] {
       case FileBody(f, _)        => BodyPublishers.ofFile(f.toFile.toPath).unit
       case StreamBody(s)         => streamToPublisher(s.asInstanceOf[streams.BinaryStream])
       case m: MultipartBody[_] =>
-        val multipartBodyPublisher = multipartBody(m.parts)
         val baseContentType = contentType.getOrElse("multipart/form-data")
-        builder.header(HeaderNames.ContentType, s"$baseContentType; boundary=${multipartBodyPublisher.getBoundary}")
-        multipartBodyPublisher.build().unit
+        if (streams == NoStreams) {
```

Comment:
alternative implementation idea: we could have a `MultipartBodyBuilder` trait, with the types of implementations: one which supports only non-streaming parts (the default), and another one,  per-streaming-type.

The trait could have methods such as `addByteArrayPart`, `addInputStreamPart` etc., and in the end it would have to produce ("somehow") a `BodyPublisher`. Maybe this would be cleaner, as then the `concatStreams` methods etc could be part of the implementations of this trait?

Reply:
I added `MultipartBodyBuilder` that accepts `Seq[Part[GenericRequestBody[_]]]` and produces `F[BodyPublisher]`. It reuses logic for cases without stream support.